lib/ansible/modules/cloud/cloudscale/cloudscale_floating_ip.py

```diff
@@ -0,0 +1,275 @@
+        single address is created.
+    choices: [ 56 ]
+  reverse_ptr:
+    description:
+      - Reverse PTR entry for this address.
+      - You cannot set a reverse PTR entry for IPv6 floating networks. Reverse PTR entries are only allowed for single addresses.
+  api_token:
+    description:
+      - cloudscale.ch API token.
+      - This can also be passed in the CLOUDSCALE_API_TOKEN environment variable.
+  api_timeout:
+    description:
+      - Timeout in seconds for calls to the cloudscale.ch API.
+    default: 30
+'''
+
+EXAMPLES = '''
+# Request a new floating IP
+- name: Request a floating IP
+  cloudscale_floating_ip:
+    ip_version: 4
+    server: 47cec963-fcd2-482f-bdb6-24461b2d47b1
+    reverse_ptr: my-server.example.com
+    api_token: xxxxxx
+  register: floating_ip
+
+# Assign an existing floating IP to a different server
+- name: Move floating IP to backup server
+  cloudscale_floating_ip:
+    ip: 192.0.2.123
+    server: ea3b39a3-77a8-4d0b-881d-0bb00a1e7f48
+    api_token: xxxxxx
+
+# Request a new floating IPv6 network
+- name: Request a floating IP
+  cloudscale_floating_ip:
+    ip_version: 6
+    prefix_length: 56
+    server: 47cec963-fcd2-482f-bdb6-24461b2d47b1
+    api_token: xxxxxx
+  register: floating_ip
+
+# Assign an existing floating network to a different server
+- name: Move floating IP to backup server
+  cloudscale_floating_ip:
+    ip: '{{ floating_ip.network | ip }}'
+    server: ea3b39a3-77a8-4d0b-881d-0bb00a1e7f48
+    api_token: xxxxxx
+
+# Release a floating IP
+- name: Release floating IP
+  cloudscale_floating_ip:
+    ip: 192.0.2.123
+    state: absent
+    api_token: xxxxxx
+'''
+
+RETURN = '''
+href:
+  description: The API URL to get details about this floating IP.
+  returned: success when state == present
+  type: string
+  sample: https://api.cloudscale.ch/v1/floating-ips/2001:db8::cafe
+network:
+  description: The CIDR notation of the network that is routed to your server.
+  returned: success when state == present
+  type: string
+  sample: 2001:db8::cafe/128
+next_hop:
+  description: Your floating IP is routed to this IP address.
+  returned: success when state == present
+  type: string
+  sample: 2001:db8:dead:beef::42
+reverse_ptr:
+  description: The reverse pointer for this floating IP address.
+  returned: success when state == present
+  type: string
+  sample: 185-98-122-176.cust.cloudscale.ch
+server:
+  description: The floating IP is routed to this server.
+  returned: success when state == present
+  type: string
+  sample: 47cec963-fcd2-482f-bdb6-24461b2d47b1
+ip:
+  description: The floating IP address or network. This is always present and used to identify floating IPs after creation.
+  returned: success
+  type: string
+  sample: 185.98.122.176
+state:
+  description: The current status of the floating IP.
+  returned: success
+  type: string
+  sample: present
+'''
+
+import os
+
+from netaddr import IPNetwork
```

Comment:
Is this module included with Python?
If not needs putting in a try block. Search other modules for `HAS_LIB`

Reply:
I'll have a look and fix this.